File: lintcode/invert_binary_tree.py
"""
Definition of TreeNode:
class TreeNode:
    def __init__(self, val):
        self.val = val
        self.left, self.right = None, None
"""
class Solution:
    # @param root: a TreeNode, the root of the binary tree
    # @return: nothing
    def invertBinaryTree(self, root):
        if root:
            root.left, root.right = self.invertBinaryTree(root.right), \
                                    self.invertBinaryTree(root.left)
        return root

File: leetcode/invert_binary_tree.js
/* Invert a binary tree.
 * Time Complexity: O(n)
 * Space Complexity: O(h), h is the height of the tree
 */
/**
 * Definition for a binary tree node.
 * function TreeNode(val) {
 *     this.val = val;
 *     this.left = this.right = null;
 * }
 */
/**
 * @param {TreeNode} root
 * @return {TreeNode}
 */
var invertTree = function(root) {
    if (root) {
        var temp = invertTree(root.right)
        root.right = invertTree(root.left)
        root.left = temp
    }
    return root
};
